### PR TITLE
fix(events): fix property expression extraction code to not break $parse

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

**Comment by erwinmombay** [comment]
@btford @SomeKittens can i get a review?

unfortunately this still doesn't perfectly cover all the cases (unless we build/copy angular's expression parser) but ng-hint should definitely not blow up the user's application first and foremost.

File: src/modules/events.js [hunk 1]
```diff
@@ -6,18 +6,43 @@
...
 
 var getFunctionNames = function(str) {
   if (typeof str !== 'string') {
     return [];
   }
-  var results = str.replace(/\s+/g, '').split(/[\+\-\/\|\<\>\^=&!%~;]/g).map(function(x) {
-    if (isNaN(+x)) {
-      if (x.match(/\w+\(.*\)$/)){
-        return x.substr(0, x.indexOf('('));
+  // There are still a bunch of corner cases here where we are'nt going to be able to handle
```
Comment (btford) [inline src/modules/events.js]: `are'nt` -> `aren't`

**Comment by btford** [comment]
LGTM

**Comment by SomeKittens** [comment]
:shipit:

**Comment by btford** [comment]
Landed as 26cfc749113a599a1e11a587091c558ee2e499b6

**Pull request closed without merging**